projects/vdk-plugins/vdk-confluence-data-source/src/vdk/plugin/confluence_data_source/data_source.py

```diff
@@ -0,0 +1,224 @@
+# Copyright 2021-2024 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+import logging
+from typing import Iterator
+from typing import List
+from typing import Optional
+
+from atlassian import Confluence
+from retrying import retry
+from vdk.plugin.data_sources.config import config_class
+from vdk.plugin.data_sources.config import config_field
+from vdk.plugin.data_sources.data_source import DataSourcePayload
+from vdk.plugin.data_sources.data_source import IDataSource
+from vdk.plugin.data_sources.data_source import IDataSourceConfiguration
+from vdk.plugin.data_sources.data_source import IDataSourceStream
+from vdk.plugin.data_sources.factory import data_source
+
+
+@config_class(
+    name="confluence-data-source-config", description="Stream Confluence page contents"
+)
+class ConfluenceContentDataSourceConfiguration(IDataSourceConfiguration):
+    confluence_url: str = config_field(description="URL of the Confluence site")
+    space_key: str = config_field(description="Key of the Confluence space")
+    username: Optional[str] = config_field(
+        description="Confluence username for authentication", default=None
+    )
+    api_token: Optional[str] = config_field(
+        description="Confluence API token for authentication", default=None
+    )
+    personal_access_token: Optional[str] = config_field(
+        description="Confluence personal access token for" " authentication",
+        default=None,
+    )
+    oauth2: Optional[dict] = config_field(
+        description="OAuth2 credentials for authentication "
+        "in dictionary format. Required keys: 'access_token',"
+        " 'access_token_secret', 'consumer_key', 'key_cert'",
+        default=None,
+    )
+    cloud: bool = config_field(
+        description="Flag indicating if the Confluence instance is" " cloud-based",
+        default=True,
+    )
+    confluence_kwargs: Optional[dict] = config_field(
+        description="Additional keyword arguments " "for the Confluence client",
```

Comment:
what are those keyword arguments? What is this "confluence client" 

As a user reading the description I'd wonder those.

(hint: link to relevant sites may be ok)